src/polymer/behavior-scanner.ts

```diff
@@ -67,11 +67,34 @@ class BehaviorVisitor implements Visitor {
   propertyHandlers: PropertyHandlers|null = null;
 
   document: JavaScriptDocument;
+
   constructor(document: JavaScriptDocument) {
     this.document = document;
   }
 
   /**
+   * Fix how comments are attached to SequenceExpressions. In espree, any
+   * leading comments to the expression are parsed as leading comments to the
+   * entire sequence and not the first expression in that sequence.
+   *
+   * For our behaviors we expect our `@polymerBehavior` comments to be the
+   * leading comments of the actual first expression, so we copy them over here
+   * before so that other enter functions can properly read them.
+   */
+  enterSequenceExpression(
+      node: estree.SequenceExpression, parent: estree.Node) {
+    node.expressions.forEach((expression, i) => {
+      if (i === 0 && !expression.leadingComments) {
+        node.leadingComments = node.leadingComments || parent.leadingComments;
```

Comment:
Damn, good point. Ideally we would have access to the complete stack of ancestors inside this visitor.

I'm not super happy with a visitor modifying the nodes that it's walking, as that means that potentially other visitors would be affected by the order in which they are run, tying us to the implementation details of the tree walking algorithm. Do you see a way to do this without modifying estree nodes?